subprojects/internal-testing/src/test/groovy/org/gradle/testing/internal/util/RetryRuleWithSetupRerunTest.groovy

```diff
@@ -16,28 +16,46 @@
 
 package org.gradle.testing.internal.util
 
+import org.gradle.util.GUtil
+import org.junit.ClassRule
 import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Shared
 import spock.lang.Specification
 
 import static org.gradle.testing.internal.util.RetryRule.retryIf
 
 class RetryRuleWithSetupRerunTest extends SuperSpecification {
 
+    private static final String ITERATION_PROPERTY_NAME = "iteration"
+    private static final String SETUP_CALL_COUNT_PROPERTY_NAME = "setupCallCount"
+
     @Rule
     RetryRule retryRule = retryIf({ t -> t instanceof IOException })
 
-    int iteration = 0
+    int iteration
 
-    int setupCallCount = 0
+    int setupCallCount
 
     def setup() {
-        setupCallCount++
+        def specificationFile = new File(temporaryFolder.root, specificationContext.currentFeature.name)
```

Comment:
I'm not sure I understand what this code is doing? Why are we writing to a properties file?

Reply:
We cannot use the field any more to count iterations, since the field is reset by the `RetryRule`. So we keep the state in a file per test method instead.

Reply:
Not even if we make it a `@Shared` field?

Reply:
A shared field would work, but how do we reset the field between different test methods?
Now that I think about it, I could use a shared `Map` from test method name to iteration count. WDYT?

Reply:
That sounds a little easier to read, yeah.

Reply:
Better now?